src/app/components/daily-review/daily-review.component.ts

```diff
@@ -1,50 +1,328 @@
+  
+    if (this.newHabitLogEntry === true) {
+      
+      Object.keys(this.habitReviewForm.controls).forEach(key => {
+        console.log(
+          "keys =",
+          key,
+          this.habitReviewForm.controls[key].value
+        );
+        habitLogs.push({
+          id: 0,
+          userId: this.currentUserId,
+          habitId: Number(key),
+          completed: this.habitReviewForm.controls[key].value,
+          date: this.currentDate
+        });
+      });
+      
+      this.accountService.addHabitCompletionLogs(habitLogs);
+
+    } else if (this.newHabitLogEntry === false) {
+     
+      console.log("form before controls = ", this.currentDateHabitLogs);
+      Object.keys(this.habitReviewForm.controls).forEach(key => {
+        this.currentDateHabitLogs.forEach((log) => {
+          if (key === log.habitId.toString()) {
+            log.completed = this.habitReviewForm.controls[key].value;
+          }
+        });
+        console.log("keys =", key, this.habitReviewForm.controls[key].value);
+
+      });
+    }
+    this.accountService.updateHabitCompletionLogs(this.currentDateHabitLogs);
+
+  }
+
+  submitWellnessRatingForm() {
+    const wellnessFormData: IWellnessRating = this.wellnessRatingForm.getRawValue();
+
+    if (this.newWellnessEntry === true) {
+
+      this.newWellnessEntry = false;
+
+      this.accountService.addWellnessRatingEntry(wellnessFormData)
+      .pipe(take(1))
+      .subscribe();
+    } else if (this.newWellnessEntry === false) {
+      console.log("new entry false");
+
+      this.accountService.getCurrentDateWellnessEntry(
+        this.currentDate,
+        this.currentUserId
+      ).
+        pipe(take(1)).
+        subscribe(wellnessEntry => {
+
+          if (wellnessEntry.length !== 0) {
+            //update the existing currentDate wellness entry with the Forms data
+            this.wellnessRecordId = wellnessEntry[0].id;
+            wellnessFormData.id = this.wellnessRecordId;
+            this.accountService.updateWellnessData(wellnessFormData).
+              pipe(take(1)).
+              subscribe(() => {
+                console.log("form update success");
+                this.alertService.success('Wellness Entry Successfully Updated');
+              });
+
+          }
+        });//end subscribe
+    }
+  }
+
+  initializeForms() {
+
+    this.habitReviewForm = this.fb.group({});
+
+    //Initialize the habit form with the users saved habits, and generate a control for each
+    this.accountService.getHabitsForCurrentUser(this.currentUserId).
+      pipe(take(1)).
+      subscribe(habits => {
+        if (habits.length !== 0) {
+          habits.forEach(habit => {
+            this.currentUserHabits.push(habit);
+            this.habitReviewForm.addControl(habit.id.toString(), new FormControl(false));
+          });
+        }
+        else {
+          this.noHabits = true;
+        }
+      });
+
+    this.wellnessRatingForm = this.fb.group({
```

Comment:
Always a good reminder that this form builder code executes before the contents of the subscribe above. Be careful doing anything in the same method expecting results from the observable. It's async. I don't think you have a problem here but it could result in one if you accidentally used anything that was set in that observable above.